### PR TITLE
Expose WordDelimiterGraphTokenFilter

### LINKED ISSUE #23104: Expose and document WordDelimiterGraphTokenFilter
In Lucene 6.5.0 the word delimiter filter has a graph version that is able to handle multi terms accurately. We should expose this new filter and document how it works like we did for the SynonymGraphFilter.
Depends on https://github.com/elastic/elasticsearch/pull/23087

### PR DESCRIPTION
**Pull request by jimczi**
This change exposes the new Lucene graph based word delimiter token filter in the analysis filters.
Unlike the `word_delimiter` this token filter named `word_delimiter_graph` correctly handles multi terms expansion at query time.

Closes #23104

### CODE REVIEW

File: core/src/test/java/org/elasticsearch/index/analysis/BaseWordDelimiterTokenFilterFactoryTestCase.java [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.env.Environment;
+import org.elasticsearch.test.ESTestCase;
+import org.elasticsearch.test.ESTokenStreamTestCase;
+
+import java.io.IOException;
+import java.io.StringReader;
+
+/**
+ * Base class to test {@link WordDelimiterGraphTokenFilterFactory}  and {@link WordDelimiterGraphTokenFilterFactory}
```
Comment (mikemccand) [inline core/src/test/java/org/elasticsearch/index/analysis/BaseWordDelimiterTokenFilterFactoryTestCase.java]: One of these shouldn't have `Graph` in it?

File: docs/reference/analysis/tokenfilters/word-delimiter-graph-tokenfilter.asciidoc [hunk 2]
```diff
@@ -0,0 +1,97 @@
+[[analysis-word-delimiter-graph-tokenfilter]]
+=== Word Delimiter Graph Token Filter
+
+experimental[]
+
+Named `word_delimiter_graph`, it Splits words into subwords and performs
```
Comment (mikemccand) [inline docs/reference/analysis/tokenfilters/word-delimiter-graph-tokenfilter.asciidoc]: Lowercase `splits` here?

**Review by mikemccand — APPROVED** [review]
LGTM; just two minor comments.  Thanks @jimczi!

**Comment by jimczi** [comment]
Thanks @mikemccand !

**Pull request merged**